### ISSUE TITLE
error creating a machine in el-capitan

### ISSUE THREAD

**Issue by BlueHotDog** [codekitchen/dinghy#90]
```
Installing fsevents_to_vm, this will require sudo
ERROR:  While executing gem ... (Errno::EPERM)
    Operation not permitted - /usr/bin/thor
/usr/local/Cellar/dinghy/HEAD/cli/dinghy/plist.rb:35:in `system!': Error with the FsEvents daemon during installing (RuntimeError)
        from /usr/local/Cellar/dinghy/HEAD/cli/dinghy/fsevents_to_vm.rb:42:in `install_if_necessary!'
        from /usr/local/Cellar/dinghy/HEAD/cli/dinghy/fsevents_to_vm.rb:15:in `up'
        from /usr/local/Cellar/dinghy/HEAD/cli/cli.rb:184:in `start_services'
        from /usr/local/Cellar/dinghy/HEAD/cli/cli.rb:78:in `up'
        from /usr/local/Cellar/dinghy/HEAD/cli/cli.rb:128:in `restart'
        from /usr/local/Cellar/dinghy/HEAD/cli/cli.rb:145:in `upgrade'
        from /usr/local/Cellar/dinghy/HEAD/cli/thor/lib/thor/command.rb:27:in `run'
        from /usr/local/Cellar/dinghy/HEAD/cli/thor/lib/thor/invocation.rb:126:in `invoke_command'
        from /usr/local/Cellar/dinghy/HEAD/cli/thor/lib/thor.rb:359:in `dispatch'
        from /usr/local/Cellar/dinghy/HEAD/cli/thor/lib/thor/base.rb:440:in `start'
        from /usr/local/bin/_dinghy_command:6:in `<main>'
```

everytime i try to create the vm

**Comment by codekitchen** [comment]
Are you on the latest version? There were a few fixes for El Capitan merged in last week.

If you are, then it looks like something unusual is happening with your OS X system ruby -- it's still trying to install gem binaries to `/usr/bin` instead of `/usr/local/bin`, which is no longer allowed with El Capitan. Do you get the same error if you run the install command outside of dinghy?

    sudo /System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/gem install --no-rdoc --no-ri fsevents_to_vm -v ~> 1.0.1

**Comment by BlueHotDog** [comment]
same exception:
```
ERROR:  While executing gem ... (Errno::EPERM)
    Operation not permitted - /usr/bin/thor
```

**Comment by BlueHotDog** [comment]
if i install it without the full path to "gem" it works..
i.e like this:
```sudo gem install --no-rdoc --no-ri fsevents_to_vm -v '~> 1.0.1' ```

**Comment by codekitchen** [comment]
It's probably a different `gem` binary, do you use a ruby version manager? I've never seen this before but this might be related, make sure you have the latest xcode: http://stackoverflow.com/a/32963315

**Comment by BlueHotDog** [comment]
Tried, nothing.

I use rbenv

**Comment by BlueHotDog** [comment]
ok, i think it's simply an rbevn related issue,
why not use $(which gem) instead of giving it a direct path?

**Comment by BlueHotDog** [comment]
which makes sense, your code doesn't allow for things like rvm/rbenv to work


Sent with Mixmax

On Thu, Oct 15, 2015 at 1:53 PM Danni Friedland <dannifriedland@gmail.com>
wrote:
Tried, nothing.

I use rbenv
On Thu, Oct 15, 2015 at 7:55 AM Brian Palmer < notifications@github.com > wrote:
It's probably a different gem binary, do you use a ruby version manager? I've never seen this before but this
might be related, make sure you have the latest xcode: http://stackoverflow.com/a/ 32963315

**Comment by codekitchen** [comment]
For the same reason that homebrew hard codes to the system ruby (see `/usr/local/Library/brew.rb`) -- it's impossible to support all the infinite versions, configurations, and installations of ruby that people could have active with rvm/rbenv/chruby, but typically we know what state the system ruby is in, because it ships with the OS.

Unfortunately it looks like that isn't true in your case, but it's not clear to me what's different.

**Comment by dfeyer** [comment]
Based on http://stackoverflow.com/questions/31972968/cant-install-gems-on-macos-x-el-capitan, don't use sudo to install gem on El Capitain, without sudo the command work fine:

```
/System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/gem install --no-rdoc --no-ri fsevents_to_vm -v '~> 1.0.1'
```

**Comment by dfeyer** [comment]
#93 Solve the issue for me can you try it @BlueHotDog ?

**Comment by thatgibbyguy** [comment]
The issue is that El Capitan introduces something called System Integrity Protection. You must remove this in order for `dinghy create` to work. Instructions for that are at https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/El_Capitan_and_Homebrew.md

Another fix would be to not hard code dinghy to use the system version of ruby and instead defer to what the user prefers for gem management. I noticed this at https://github.com/codekitchen/dinghy/blob/latest/cli/dinghy/fsevents_to_vm.rb#L39 - please let me know if it's desirable for me to fork and create a fix for this.

**Comment by dfeyer** [comment]
I think disabling System Integrity Protection is not a valid option on the long run. If you have trouble with a security device you don't remove it you try to configure it correctly or adapt to live with it ;)

Installing the gem in /usr/local/bin if OK, it's in the PATH of 100% of users.

About remove hardcoded path to gem, check the previous comment from @codekitchen it's not an option.

**Comment by codekitchen** [comment]
Yes, please don't disable System Integrity Protection just to use dinghy. I'll cry if you do.

What does running `/System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/gem env` on el capitan give for the install paths for people? I've tried it on 4 machines so far, and all 4 have returned:

```
  - INSTALLATION DIRECTORY: /Library/Ruby/Gems/2.0.0
  - EXECUTABLE DIRECTORY: /usr/local/bin
```

I'm guessing the original reporter will have `/usr/bin` reported for their executable directory. I'm not sure why. #93 might be an OK solution though.

**Comment by thatgibbyguy** [comment]
You can reenable SIP after you're done. I'm just repeating what Homebrew is recommending, it worked for me and so far is the only solution in this thread. If another solution is provided, I'll delete my comment.

**Comment by codekitchen** [comment]
If anybody else who is seeing this problem can test out https://github.com/codekitchen/dinghy/pull/93 I'd much appreciate it, as I've yet to find an El Capitan machine that I have access to that exhibits this behavior. Thanks!

**Comment by BlueHotDog** [comment]
how do i test that?

**Comment by BlueHotDog** [comment]
and yeah, my EXECUTABLE dir is /usr/bin

**Comment by BlueHotDog** [comment]
ok, tried it, i think, via:
```
dinghy destroy
brew reinstall --HEAD https://github.com/dfeyer/dinghy/raw/patch-1/dinghy.rb
dinghy create --provider virtualbox
```

same exception

what is wrong with my machine :(

**Comment by codekitchen** [comment]
Ah that will still install the `master` branch because of how homebrew references branches. To try it out, clone the dinghy repo locally, pull down that PR into your cloned repo, and then run `./bin/dinghy create` instead of `dinghy create`.

**Comment by BlueHotDog** [comment]
cool

**Comment by BlueHotDog** [comment]
what i did:
```
git clone https://github.com/dfeyer/dinghy
cd dinghy
git checkout patch-1
./bin/dinghy create
```
what i received:
```
/Library/Ruby/Site/2.0.0/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- thor (LoadError)
```

**Comment by codekitchen** [comment]
Oh you also need to run `git submodule update --init` to pull down the git submodules. I've been working on fleshing out a https://github.com/codekitchen/dinghy/blob/master/CONTRIBUTING.md document to help people get set up with all this, I'll add that step.

**Comment by BlueHotDog** [comment]
ok, that worked!!!

**Comment by BlueHotDog** [comment]
but now the FSEV is marked as not running :(

**Comment by thatgibbyguy** [comment]
Try `dinghy destroy` then `dinghy create --provider=virtualbox` and let us know if it installs the fsEvents gem.

**Comment by BlueHotDog** [comment]
It does.. But its not running

**Comment by BlueHotDog** [comment]
Do you guys mind entering gitter or something? So we can try and resolve this?

**Comment by codekitchen** [comment]
I left a comment on the PR, I don't think it's a complete solution. I haven't heard back from @dfeyer though so I'll try to find some time to code it up in the next couple days. It might be best to just tell rubygems to install the binary right alongside dinghy's other files. Apologies in advance if it takes me a while to get to it, things are crazy busy right now.

**Comment by BlueHotDog** [comment]
Nm guys, ive just moved to AZK, thanks for trying to solve this though

**Comment by codekitchen** [comment]
OK, I've released dinghy 4.0.5 that should fix this issue across the board, it explicitly installs the binary to `/usr/local/bin` and looks for it there on bootup. If anybody hits issues still, let me know.
